src/storage/endorsements_repository_inmem.cpp

```diff
@@ -0,0 +1,58 @@
+#include "veriblock/storage/endorsements_repository_inmem.hpp"
+
+using namespace VeriBlock;
+
+void EndorsementsRepositoryInmem::put(
+    const VbkPopTx& popTx, const VbkBlock::hash_t& containingBlockHash) {
+  VbkBlock::hash_t endorsedBlockHash = popTx.publishedBlock.getHash();
+  std::string popTxHash = popTx.getHash().asString();
+
+  std::vector<Endorsement>& endrs = endorsed_map[endorsedBlockHash];
+  auto endorsement = endrs.begin();
+  for (; endorsement != endrs.end(); ++endorsement) {
+    if (endorsement->endorsementKey == popTxHash) {
+      endorsement->containingVbkHashes.push_back(containingBlockHash);
+      break;
+    }
+  }
+  if (endrs.empty()) {
+    endorsed_map[endorsedBlockHash] = {{
+        popTx.getHash().asString(),
+        popTx.publishedBlock.getHash(),
+        {containingBlockHash},
+        popTx.blockOfProof.getHash(),
+        popTx.address.toString(),
```

Comment:
we store address, but do we use it?

Reply:
as I remember, no.

Reply:
remove it, then

Reply:
No I have mistaken, we will able to use it in the pop reward algorithm